service/src/main/kotlin/no/nav/su/se/bakover/service/behandling/BehandlingServiceImpl.kt

```diff
@@ -0,0 +1,171 @@
+import no.nav.su.se.bakover.domain.Behandling
+import no.nav.su.se.bakover.domain.Saksbehandler
+import no.nav.su.se.bakover.domain.behandling.Behandlingsinformasjon
+import no.nav.su.se.bakover.domain.beregning.Fradrag
+import no.nav.su.se.bakover.domain.oppdrag.NyUtbetaling
+import no.nav.su.se.bakover.domain.oppdrag.simulering.SimuleringClient
+import no.nav.su.se.bakover.domain.oppdrag.simulering.SimuleringFeilet
+import no.nav.su.se.bakover.domain.oppdrag.utbetaling.UtbetalingPublisher
+import no.nav.su.se.bakover.domain.oppgave.KunneIkkeOppretteOppgave
+import no.nav.su.se.bakover.domain.oppgave.OppgaveClient
+import no.nav.su.se.bakover.domain.oppgave.OppgaveConfig
+import no.nav.su.se.bakover.service.sak.SakService
+import no.nav.su.se.bakover.service.søknad.FantIkkeSøknad
+import no.nav.su.se.bakover.service.søknad.SøknadService
+import no.nav.su.se.bakover.service.utbetaling.UtbetalingService
+import java.time.LocalDate
+import java.util.UUID
+
+internal class BehandlingServiceImpl(
+    private val behandlingRepo: BehandlingRepo,
+    private val hendelsesloggRepo: HendelsesloggRepo,
+    private val beregningRepo: BeregningRepo,
+    private val oppdragRepo: OppdragRepo,
+    private val simuleringClient: SimuleringClient,
+    private val utbetalingService: UtbetalingService, // TODO use services or repos? probably services
+    private val oppgaveClient: OppgaveClient,
+    private val utbetalingPublisher: UtbetalingPublisher,
+    private val søknadService: SøknadService,
+    private val sakService: SakService
+) : BehandlingService {
+    override fun hentBehandling(behandlingId: UUID): Either<FantIkkeBehandling, Behandling> {
+        return behandlingRepo.hentBehandling(behandlingId)?.right() ?: FantIkkeBehandling.left()
+    }
+
+    override fun underkjenn(
+        begrunnelse: String,
+        attestant: Attestant,
+        behandling: Behandling
+    ): Either<Behandling.KunneIkkeUnderkjenne, Behandling> {
+        return behandling.underkjenn(begrunnelse, attestant)
+            .map {
+                hendelsesloggRepo.oppdaterHendelseslogg(it.hendelseslogg)
+                behandlingRepo.hentBehandling(it.id)!!
+            }
+    }
+
+    override fun oppdaterBehandlingsinformasjon(
+        behandlingId: UUID,
+        behandlingsinformasjon: Behandlingsinformasjon
+    ): Behandling {
+        val beforeUpdate = behandlingRepo.hentBehandling(behandlingId)!!
+        beregningRepo.slettBeregningForBehandling(behandlingId)
+        val updated = behandlingRepo.oppdaterBehandlingsinformasjon(
+            behandlingId,
+            beforeUpdate.behandlingsinformasjon().patch(behandlingsinformasjon)
+        )
+        // TODO fix weirdness for internal state
+        val status = updated.oppdaterBehandlingsinformasjon(behandlingsinformasjon).status()
+        behandlingRepo.oppdaterBehandlingStatus(behandlingId, status)
+        return behandlingRepo.hentBehandling(behandlingId)!!
```

Comment:
Så vi henter behandling 3 ganger hvorav en er skjult, istedet for å stole på den interne modellen?

Reply:
Ref diskusjonene vi har hatt angående dette. Hvem har ansvaret for hva og når? Dette får vi komme tilbake til. Burde vel ha en større diskusjon om hvordan dette bør være.